### PR TITLE
Store number of points in an std::array.

### LINKED ISSUE #214: Missing detectors in R3BMCTrack
To reproduce: run test file `macros/r3b/run_sim.C()` directly
Tested with 3b073bf1ee7cc8505ae6c35fa4f3aeaa4aa5a702 (before the macros where removed from the main repository)

The detectors added in
https://github.com/R3BRootGroup/R3BRoot/commit/90c91987d2b833f347f4f77d9e0a43269ac919ed#diff-7fe5d291e474bea07c4c9def6abcc1f7

are missing in `R3BMCTrack::SetNPoints` and `R3BMCTrack::GetNPoints`, which leads to many warning messages during simulation:
```
-E- R3BMCTrack::SetNPoints: Unknown detector ID 17
-E- R3BMCTrack::SetNPoints: Unknown detector ID 18
-E- R3BMCTrack::SetNPoints: Unknown detector ID 22
-E- R3BMCTrack::SetNPoints: Unknown detector ID 23
-E- R3BMCTrack::SetNPoints: Unknown detector ID 24
-E- R3BMCTrack::SetNPoints: Unknown detector ID 25
```

I would highly recommend ditching the current implementation of `fNPoints`, a overcomplicated Bitvector, in favor of, e.g., a std::array with size of the enum `DetectorId`.

Also, I don't know what this is part in `R3BMCStack` does, but is seems somewhat wasteful. Maybe just store the whole `fPointsMap` in `R3BMCTrack`?
```C++
      for (Int_t iDet=kREF; iDet<kLAST; iDet++) {
        pair<Int_t, Int_t> a(iPart, iDet);
        track->SetNPoints(iDet, fPointsMap[a]);
      }
```

### PR DESCRIPTION
**Pull request by kresan**
resolves #214

### CODE REVIEW

File: r3bdata/R3BMCTrack.cxx [hunk 1]
```diff
@@ -20,9 +22,12 @@ R3BMCTrack::R3BMCTrack()
     , fStartY(0.)
     , fStartZ(0.)
     , fStartT(0.)
-    , fNPoints(0)
     , fMass(0.)
 {
+    for(auto & x : fNPoints)
```
Comment (janmayer) [inline r3bdata/R3BMCTrack.cxx]: To initialize with zeroes, you can use `{}`, i.e., `fNPoints({})`

```
// Example program
#include <iostream>
#include <array>

class Bar {
    std::array<int, 20> foo;

public:    
    Bar() 
    //: foo({}) // Compare output w/ and w/o
    {
        for (const auto& x : foo){
           std::cout << x << "\n";
        }        
    }
};

int main()
{
    Bar bar;
}
```

File: r3bdata/R3BMCTrack.cxx [hunk 2]
```diff
@@ -46,17 +51,9 @@ R3BMCTrack::R3BMCTrack(Int_t pdgCode,
...
 {
-    if (nPoints >= 0)
-    {
-        fNPoints = nPoints;
-    }
-    else
-    {
-        fNPoints = 0;
-    }
+    fNPoints = nPoints;
```
Comment (janmayer) [inline r3bdata/R3BMCTrack.cxx]: You can also put this in the initializer list

File: r3bdata/R3BMCTrack.cxx [hunk 3]
```diff
@@ -74,6 +71,7 @@ R3BMCTrack::R3BMCTrack(const R3BMCTrack& right)
     , fNPoints(right.fNPoints)
     , fMass(right.fMass)
 {
+    fNPoints = right.fNPoints;
```
Comment (janmayer) [inline r3bdata/R3BMCTrack.cxx]: You can also put this in the initializer list

File: r3bdata/R3BMCTrack.cxx [hunk 4]
```diff
@@ -101,6 +98,11 @@ R3BMCTrack::R3BMCTrack(TParticle* part, Int_t fMC)
         // G4
         fStartT = part->T();
     }
+
```
Comment (janmayer) [inline r3bdata/R3BMCTrack.cxx]: Same as above.

**Comment by kresan** [comment]
@janmayer: requested changes are in

**Pull request merged**